debug_utils/sierra-emu/src/vm/int_range.rs

```diff
@@ -0,0 +1,95 @@
+use cairo_lang_sierra::{
+    extensions::{
+        core::{CoreLibfunc, CoreType},
+        lib_func::SignatureOnlyConcreteLibfunc,
+        range::IntRangeConcreteLibfunc,
+        ConcreteLibfunc,
+    },
+    program_registry::ProgramRegistry,
+};
+use num_bigint::BigInt;
+use smallvec::smallvec;
+
+use crate::{
+    utils::{get_numeric_args_as_bigints, get_value_from_integer},
+    Value,
+};
+
+use super::EvalAction;
+
+pub fn eval(
+    registry: &ProgramRegistry<CoreType, CoreLibfunc>,
+    selector: &IntRangeConcreteLibfunc,
+    args: Vec<Value>,
+) -> EvalAction {
+    match selector {
+        IntRangeConcreteLibfunc::TryNew(info) => eval_try_new(registry, info, args),
+        IntRangeConcreteLibfunc::PopFront(info) => eval_pop_front(registry, info, args),
+    }
+}
+
+fn eval_try_new(
+    registry: &ProgramRegistry<CoreType, CoreLibfunc>,
+    info: &SignatureOnlyConcreteLibfunc,
+    args: Vec<Value>,
+) -> EvalAction {
+    let range_check @ Value::Unit: Value = args[0].clone() else {
+        panic!()
+    };
+    let [x, y]: [BigInt; 2] = get_numeric_args_as_bigints(&args[1..]).try_into().unwrap();
+
+    let int_ty = registry.get_type(&info.param_signatures()[1].ty).unwrap();
+
+    let is_valid_range = x < y;
+
+    // if x >= y then the range is not valid and we return [y, y) (empty range)
+    let range = if is_valid_range {
+        let x = get_value_from_integer(registry, int_ty, x);
+        let y = get_value_from_integer(registry, int_ty, y);
+
+        Value::IntRange {
+            x: Box::new(x),
+            y: Box::new(y),
+        }
+    } else {
+        let y = get_value_from_integer(registry, int_ty, y);
+
+        Value::IntRange {
+            x: Box::new(y.clone()),
+            y: Box::new(y),
+        }
+    };
+
+    EvalAction::NormalBranch(is_valid_range as usize, smallvec![range_check, range])
```

Comment:
```suggestion

    // if x >= y then the range is not valid and we return [y, y) (empty range)
    if x < y {
        let x = get_value_from_integer(registry, int_ty, x);
        let y = get_value_from_integer(registry, int_ty, y);

        EvalAction::NormalBranch(
            0,
            smallvec![
                range_check,
                Value::IntRange {
                    x: Box::new(x),
                    y: Box::new(y),
                }
            ],
        )
    } else {
        let y = get_value_from_integer(registry, int_ty, y);

        EvalAction::NormalBranch(
            1,
            smallvec![
                range_check,
                Value::IntRange {
                    x: Box::new(y.clone()),
                    y: Box::new(y),
                }
            ],
        )
    }
```

So that it follows the same style than this: https://github.com/lambdaclass/cairo_native/blob/502923dc0d1ab7f98fa478326c16c15b54b62dab/debug_utils/sierra-emu/src/vm/int_range.rs#L77-L94

Reply:
Done